Lib/pstats.py

```diff
@@ -776,4 +803,41 @@ def postcmd(self, stop, line):
     except KeyboardInterrupt:
         pass
 
+
+def _build_minimal_path_by_full_path(paths):
```

Comment:
In my view, I find this function too complex for what are we going to obtain using it. I had to read it 3 or 4 times to understand what is doing and how. Even if we could simplify it, I think it may be too much.

Idea: maybe we could just use the full path and then remove the common part of all the files. This will simplify this code considerably and will still have the advantage of giving more information that the filename.

Reply:
Yes, well.. that's what I said was going to be the result when you asked for it in the first place :)

Just removing the common prefix won't work at all, because some of those lines will be in the standard library and some in your project. How do we handle that? Remove 2 common prefixes? I think the code is getting much too complex again....